services/src/test/java/org/keycloak/vault/PlainTextVaultProviderTest.java

```diff
@@ -0,0 +1,80 @@
+package org.keycloak.vault;
+
+import org.junit.Test;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Tests for {@link PlainTextVaultProvider}.
+ *
+ * @author Sebastian Łaskawiec
+ */
+public class PlainTextVaultProviderTest {
+
+    @Test
+    public void shouldObtainSecret() throws Exception {
+        //given
+        PlainTextVaultProvider provider = new PlainTextVaultProvider(Scenario.EXISTING.getPath());
+
+        //when
+        VaultRawSecret secret1 = provider.obtainSecret("key1");
+
+        //then
+        assertNotNull(secret1);
+        assertNotNull(secret1.getRawSecret().get());
+        assertEquals("secret1", new String(secret1.getRawSecret().get().array()));
+    }
+
+    @Test
+    public void shouldReturnEmptyOptionalOnMissingSecret() throws Exception {
+        //given
+        PlainTextVaultProvider provider = new PlainTextVaultProvider(Scenario.EXISTING.getPath());
+
+        //when
+        VaultRawSecret secret = provider.obtainSecret("non-existing-key");
+
+        //then
+        assertNotNull(secret);
+        assertFalse(secret.getRawSecret().isPresent());
+    }
+
+    @Test
+    public void shouldOperateOnNonExistingVaultDirectory() throws Exception {
+        //given
+        PlainTextVaultProvider provider = new PlainTextVaultProvider(Scenario.NON_EXISTING.getPath());
+
+        //when
+        VaultRawSecret secret = provider.obtainSecret("non-existing-key");
+
+        //then
+        assertNotNull(secret);
+        assertFalse(secret.getRawSecret().isPresent());
+    }
+
+    @Test
+    public void shouldReflectChangesInASecretFile() throws Exception {
+        //given
+        Path temporarySecretFile = Files.createTempFile("vault", null);
+        Path vaultDirectory = temporarySecretFile.getParent();
+        String secretName = temporarySecretFile.getFileName().toString();
+
+        PlainTextVaultProvider provider = new PlainTextVaultProvider(vaultDirectory);
+
+        //when
+        Files.write(temporarySecretFile, "secret1".getBytes());
+        VaultRawSecret secret1 = provider.obtainSecret(secretName);
+
+        //by default it truncates and overrides content
+        Files.write(temporarySecretFile, "secret2".getBytes());
+        VaultRawSecret secret2 = provider.obtainSecret(secretName);
+
+        //then
+        assertEquals("secret1", new String(secret1.getRawSecret().get().array()));
```

Comment:
Prefer hamcrest methods like `assertThat(new String(secret1.getRawSecret().get().array()), is("secret1"));

Reply:
I created a custom assertion (`SecretContains`) to make it a bit easier to read.